File: M3-Juicy-Merge/game/tools/package-playgama.py
import sys
from pathlib import Path
from zipfile import ZipFile, ZIP_DEFLATED

dist_dir = Path(__file__).resolve().parent.parent / "dist"
build_dir = Path(__file__).resolve().parent.parent.parent / "build"
build_dir.mkdir(parents=True, exist_ok=True)
out_zip = build_dir / "juicy-merge-playgama.zip"

if not (dist_dir / "index.html").exists():
    print("Error: dist/index.html does not exist. Run npm run build:web first.")
    sys.exit(1)

with ZipFile(out_zip, "w", ZIP_DEFLATED) as zf:
    for f in dist_dir.rglob("*"):
        if f.is_file():
            rel = f.relative_to(dist_dir).as_posix()
            # Do not include Reddit server or devvit client bundles in Playgama zip
            if not rel.startswith("server") and not rel.startswith("client"):
                zf.write(f, rel)

size_mb = out_zip.stat().st_size / (1024 * 1024)
print(f"Playgama package created successfully: {out_zip} ({size_mb:.2f} MB)")

File: M3-Juicy-Merge/game/tools/package-playgama.mjs
import fs from 'node:fs';
import path from 'node:path';
import archiver from 'archiver';

const distDir = path.resolve(import.meta.dirname, '../dist');
const buildDir = path.resolve(import.meta.dirname, '../../build');
const outZip = path.resolve(buildDir, 'juicy-merge-playgama.zip');

if (!fs.existsSync(distDir)) {
  console.error('Error: dist/ does not exist. Run npm run build:web first.');
  process.exit(1);
}

if (!fs.existsSync(buildDir)) {
  fs.mkdirSync(buildDir, { recursive: true });
}

const output = fs.createWriteStream(outZip);
const archive = archiver('zip', { zlib: { level: 9 } });

output.on('close', () => {
  console.log(`✅ Playgama bundle created: ${outZip} (${(archive.pointer() / 1024 / 1024).toFixed(2)} MB)`);
});

archive.on('error', (err) => {
  throw err;
});

archive.pipe(output);

// Add index.html, playgama-bridge-config.json, assets/, raw/
archive.file(path.join(distDir, 'index.html'), { name: 'index.html' });
if (fs.existsSync(path.join(distDir, 'playgama-bridge-config.json'))) {
  archive.file(path.join(distDir, 'playgama-bridge-config.json'), { name: 'playgama-bridge-config.json' });
}
if (fs.existsSync(path.join(distDir, 'assets'))) {
  archive.directory(path.join(distDir, 'assets'), 'assets');
}
if (fs.existsSync(path.join(distDir, 'raw'))) {
  archive.directory(path.join(distDir, 'raw'), 'raw');
}

await archive.finalize();
